### PR TITLE
feat: added integration tests

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**

### CODE REVIEW

File: internal/controller/http/v1/devices.go [hunk 1]
```diff
@@ -15,13 +16,15 @@ type deviceRoutes struct {
...
+var ErrValidationDevices = dto.NotValidError{Console: consoleerrors.CreateConsoleError("ProfileAPI")}
+
 func newDeviceRoutes(handler *gin.RouterGroup, t devices.Feature, l logger.Interface) {
 	r := &deviceRoutes{t, l}
 
 	h := handler.Group("/devices")
 	{
 		h.GET("", r.get)
-		h.GET("stats", r.getStats)
+		h.GET("stats", r.getStats) // remove unnecessary trailing newline
```
Comment (rsdmike) [inline internal/controller/http/v1/devices.go:27]: remove comment

File: pkg/db/sql.go [hunk 2]
```diff
@@ -12,15 +11,16 @@ import (
...
 )
 
 const (
 	_defaultMaxPoolSize  = 1
 	_defaultConnAttempts = 10
 	_defaultConnTimeout  = time.Second
 
-	UniqueViolation = "23505"
+	UniqueViolation   = "23505"
+	EmUniqueViolation = "2067"
```
Comment (rsdmike) [inline pkg/db/sql.go:23]: what does Em stand for?

File: internal/usecase/sqldb/domains.go [hunk 3]
```diff
@@ -147,7 +147,7 @@ func (r *DomainRepo) GetByName(_ context.Context, domainName, tenantID string) (
 			"tenant_id",
 		).
 		From("domains").
-		Where("name = ? AND tenant_id = ?", domainName, tenantID).
+		Where("LOWER(name) = LOWER(?) AND tenant_id = ?", domainName, tenantID).
```
Comment (rsdmike) [inline internal/usecase/sqldb/domains.go:150]: did you test this with both embedded and postgres?

File: internal/usecase/profiles/usecase.go [hunk 4]
```diff
@@ -108,9 +118,38 @@ func (uc *UseCase) Delete(ctx context.Context, profileName, tenantID string) err
...
+		}
+	}
+
+	return nil
+}
+
 func (uc *UseCase) Update(ctx context.Context, d *dto.Profile) (*dto.Profile, error) {
 	d1 := uc.dtoToEntity(d)
 
+	err := uc.WifiProfileExistsfunc(ctx, d, "update")
```
Comment (rsdmike) [inline internal/usecase/profiles/usecase.go:148]: why does this have func in the name? of course its a func....

File: internal/usecase/devices/repo.go [hunk 5]
```diff
@@ -127,6 +144,9 @@ func (uc *UseCase) Update(ctx context.Context, d *dto.Device) (*dto.Device, erro
 	}
 
 	d2 := uc.entityToDTO(updateDevice)
+	if updateDevice.Tags == "" {
```
Comment (rsdmike) [inline internal/usecase/devices/repo.go:147]: thanks for fixing this 👍

File: internal/controller/http/v1/error.go [hunk 6]
```diff
@@ -19,35 +20,95 @@ type response struct {
...
+
+		return
+	case errors.As(err, &dbErr):
+		dbErrorHandle(c, dbErr)
+
+		return
+	case errors.As(err, &amtErr):
+		amtErrorHandle(c, amtErr)
+
+		return
```
Comment (rsdmike) [inline internal/controller/http/v1/error.go:57]: why did a return become necessary for all these case statements?

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: internal/controller/http/v1/devices.go (hunk 1 above)
Comment (madhavilosetty-intel) [reply internal/controller/http/v1/devices.go:27]: done

File: internal/usecase/profiles/usecase.go (hunk 4 above)
Comment (madhavilosetty-intel) [reply internal/usecase/profiles/usecase.go:148]: done

File: internal/controller/http/v1/error.go (hunk 6 above)
Comment (madhavilosetty-intel) [reply internal/controller/http/v1/error.go:57]: done

File: pkg/db/sql.go (hunk 2 above)
Comment (madhavilosetty-intel) [reply pkg/db/sql.go:23]: done

File: internal/usecase/sqldb/domains.go (hunk 3 above)
Comment (madhavilosetty-intel) [reply internal/usecase/sqldb/domains.go:150]: Yeah tested with both embedded and postgres.

**Review by rsdmike — APPROVED** [review]

**Pull request merged**